### PR TITLE
[RL] Specify list of available classes

### PR DESCRIPTION
**Pull request by Alchav**
@Berserker66 suggested this idea soon after RL was implemented and I really wanted it.
@ThePhar

### CODE REVIEW

**Comment by Berserker66** [comment]
This seems like a perfect place to make use of the VerifyKeys Mixin https://github.com/ArchipelagoMW/Archipelago/blob/e0628ec6c9adb8d6bb6516649b91b6980bde4285/Options.py#L264

Also minor Optimization would be to use OptionSet instead of OptionList

File: worlds/rogue-legacy/__init__.py [hunk 1]
```diff
@@ -93,37 +93,43 @@ def generate_basic(self):
...
-            *self._create_items(ItemName.progressive_knave),
-            *self._create_items(ItemName.progressive_shinobi),
-            *self._create_items(ItemName.progressive_miner),
-            *self._create_items(ItemName.progressive_lich),
-            *self._create_items(ItemName.progressive_spellthief),
-        ]
-
-        # Remove one of our starting classes from the item pool.
+        classes = self.world.available_classes[self.player]
+        if "Dragon" in classes:
```
[the remark this answers is not in the document]
Comment (Ijwu) [reply worlds/rogue-legacy/__init__.py:97]: Fair enough, resolving this as I have no further input.

**Review by Ijwu — APPROVED** [review]
LGTM

**Comment by Berserker66** [comment]
well, still needs to be actually implemented into the randomizer.

**Comment by Alchav** [comment]
> well, still needs to be actually implemented into the randomizer.

This PR requires no changes to the game client. It just changes which classes are in the item pool.

**Comment by ThePhar** [comment]
Looks fine to me. Might make some tweaks to the formatting later when I'm overhauling the generation function for 0.9, but we can roll this in as an additional option as it doesn't need any changes to the client.

👍

**Pull request merged**